Fix: Responses API now respects global request_timeout

colesmcintosh opened the pull request:
## Title

Fix: Responses API now respects global request_timeout

## Relevant issues

Fixes LIT-272

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem

## Type

🐛 Bug Fix

## Changes

This PR fixes an issue where the responses API endpoints were not respecting the global `request_timeout` configuration value and were timing out after 30 seconds instead.

### Problem
- The responses API was timing out after 30 seconds even when `request_timeout` in `litellm_settings` was set to a larger value
- This was inconsistent with other API endpoints that properly respect the global timeout

### Solution
- Modified `ProxyBaseLLMRequestProcessing.base_process_llm_request` to check for `litellm.request_timeout` when `user_request_timeout` is None
- This ensures the responses API uses the configured global timeout value, matching the behavior of other endpoints

### Testing
Added comprehensive tests in `tests/proxy_unit_tests/test_responses_api_timeout.py`:
- Test that `request_timeout` from config is properly propagated
- Test that explicit timeouts take precedence over global settings
- Test the timeout fallback logic in `ProxyBaseLLMRequestProcessing`

All tests pass:
```
tests/proxy_unit_tests/test_responses_api_timeout.py::test_common_request_processing_explicit_timeout PASSED [ 25%]
tests/proxy_unit_tests/test_responses_api_timeout.py::test_common_request_processing_uses_litellm_timeout PASSED [ 50%]
tests/proxy_unit_tests/test_responses_api_timeout.py::test_responses_api_timeout_from_config PASSED [ 75%]
tests/proxy_unit_tests/test_responses_api_timeout.py::test_responses_api_timeout_not_overwritten PASSED [100%]
```

colesmcintosh committed:
Fix: Responses API now respects global request_timeout (LIT-272)

- Modified ProxyBaseLLMRequestProcessing to use litellm.request_timeout when user_request_timeout is None
- Added comprehensive tests to verify timeout behavior in responses API
- Ensures consistency with other API endpoints that already respect global timeout settings

This fixes the issue where responses API was timing out at 30 seconds instead of respecting the configured global request_timeout value.

ishaan-jaff commented on litellm/proxy/common_request_processing.py:
```diff
@@ -303,6 +301,13 @@ async def common_processing_pre_call_logic(
             self.data["temperature"] = user_temperature
         if user_request_timeout:
             self.data["request_timeout"] = user_request_timeout
+        elif (
```
this is the wrong place. It should be on responses/main.py

ishaan-jaff reviewed (changes requested):
reviewed

colesmcintosh committed:
Merge branch 'BerriAI:main' into lit-272-responses-api-not-respecting-global-request-timeout

colesmcintosh committed:
fix: Set user_request_timeout from litellm.request_timeout in load_config

Previously, the responses API was not respecting the global request_timeout
set in litellm_settings. This fix ensures that when request_timeout is
configured in litellm_settings, it properly sets the global user_request_timeout
variable after the config is loaded.

The fix was moved from common_request_processing.py to proxy_server.py's
load_config method based on code review feedback, ensuring the timeout
is set at the configuration loading stage rather than during request processing.

colesmcintosh committed:
Merge branch 'BerriAI:main' into lit-272-responses-api-not-respecting-global-request-timeout

colesmcintosh committed:
Merge branch 'BerriAI:main' into lit-272-responses-api-not-respecting-global-request-timeout

colesmcintosh committed:
Merge branch 'BerriAI:main' into lit-272-responses-api-not-respecting-global-request-timeout

colesmcintosh committed:
Merge branch 'BerriAI:main' into lit-272-responses-api-not-respecting-global-request-timeout
